app/discovery/discovery.go

```diff
@@ -37,6 +37,7 @@ type URLMapper struct {
 	PingURL      string
 	MatchType    MatchType
 	RedirectType RedirectType
+	KeepHost     *bool
```

Comment:
is there a reason why this is a reference? I guess the goal is to have three-state bool here, but why? Default false for the normal bool should be fine i guess

Reply:
The primary reason we use a reference here, and in other discovery providers, is to enable the maintenance of a three-state (true/false/unset) value. If the value is not set, we'll default to the value specified in the global settings.

Reply:
got it, this is both on the global level as well on the provider level

One more thing - you don't have any tests for proxy.go changes, could you pls add it?

Reply:
sure